rust/private/repository_utils.bzl

```diff
@@ -463,12 +463,20 @@ def produce_tool_suburl(tool_name, target_triple, version, iso_date = None):
         target_triple: The rust-style target triple of the tool
         version: The version of the tool among "nightly", "beta', or an exact version.
         iso_date: The date of the tool (or None, if the version is a specific version).
-    """
 
+    Returns:
+        str: The fully qualified url path for the specified tool.
+    """
+    url = ""
     if iso_date:
-        return "{}/{}-{}-{}".format(iso_date, tool_name, version, target_triple)
-    else:
-        return "{}-{}-{}".format(tool_name, version, target_triple)
+        url += "{}/".format(iso_date)
+    if tool_name:
+        url += tool_name
+    if version:
+        url += "-{}".format(version)
```

Comment:
What do you think about alternative code (not tested):

```py
result = "-".join([e for e in [tool_name, target_triple,version] if e])
if iso_date:
  return iso_date + "/" + result
else:
  return result
```  

Combining += and format is imho confusing and cumbersome.

Reply:
Done.